app/services/description_generator.rb

```diff
@@ -146,5 +147,18 @@ def contributor(work_form_contributor)
       ]
     }
   end
+
+  sig { returns(T::Array[T::Hash[String, T::Array[T::Hash[String, String]]]]) }
+  def related_resource
+    work.related_links.map do |rel_link|
```

Comment:
I wonder if some of this could move into a new RelatedLink#to_h method, but it's not a big deal. Also, we don't (yet?) do this in other models. This way, each model could take a little more responsibility for its own representation, and the description generator can handle the formatting external to the model? I could go either way on it. Would like to hear what you think, though.

Reply:
i like this idea, though i'd also worry slightly about the `#to_h` representation being very cocina specific... but maybe that's ok?  it does seem like the main use of a model's hash representation would be for translation to cocina representation.  

looking at this example:  https://github.com/sul-dlss-labs/cocina-descriptive-metadata/pull/115/files

i'm not sure that i'd expect such a cocina specific represenation from `#to_h`.

pref for starting that here in this PR, vs doing this one case in a follow on PR, vs updating all of the hash building in this class in one fell swoop in a separate PR?

could also have a `#to_cocina` method instead of using the `#to_h` method.

Reply:
`#to_cocina` WFM too. Opinions, co-TL, @jcoyne ?

Reply:
i'd prefer `#to_cocina` or even `#to_cocina_hash` or whatever, but wondering if this refactoring should get priority over other work in ready column ...

Reply:
Like I said, not a big deal, so not a high priority.